diff -ur r8168-8.048.02/src/r8168_n.c r8168-8.048.02-256/src/r8168_n.c
--- r8168-8.048.02/src/r8168_n.c	2021-02-06 21:15:28.271768114 +0100
+++ r8168-8.048.02-256/src/r8168_n.c	2021-02-06 21:15:11.085159309 +0100
@@ -24389,11 +24389,11 @@
         tp->advertising = advertising_mode;
 
         tp->max_jumbo_frame_size = rtl_chip_info[tp->chipset].jumbo_frame_sz;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
         /* MTU range: 60 - hw-specific max */
         dev->min_mtu = ETH_ZLEN;
         dev->max_mtu = tp->max_jumbo_frame_size;
-#endif //LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
+#endif //LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
         tp->eee_enabled = eee_enable;
         tp->eee_adv_t = MDIO_EEE_1000T | MDIO_EEE_100TX;
 }
@@ -27421,12 +27421,12 @@
         int ret = 0;
         unsigned long flags;
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4,10,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
         if (new_mtu < ETH_ZLEN)
                 return -EINVAL;
         else if (new_mtu > tp->max_jumbo_frame_size)
                 new_mtu = tp->max_jumbo_frame_size;
-#endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,10,0)
+#endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
 
         spin_lock_irqsave(&tp->lock, flags);
         dev->mtu = new_mtu;
@@ -28798,7 +28798,7 @@
                 }
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
                 if (RTL_NETIF_RX_COMPLETE(dev, napi, work_done) == FALSE) return RTL_NAPI_RETURN_VALUE;
 #else
                 RTL_NETIF_RX_COMPLETE(dev, napi, work_done);
