diff -ru r8168-8.052.01/src/r8168_n.c r8168-8.052.01.strcpy/src/r8168_n.c
--- r8168-8.052.01/src/r8168_n.c	2023-09-26 16:26:06.000000000 +0200
+++ r8168-8.052.01.strcpy/src/r8168_n.c	2024-03-17 00:11:58.529966908 +0100
@@ -6293,7 +6293,7 @@
         info->eedump_len = tp->eeprom_len;
         BUILD_BUG_ON(sizeof(info->fw_version) < sizeof(rtl_fw->version));
         if (rtl_fw)
-                strlcpy(info->fw_version, rtl_fw->version,
+                strscpy(info->fw_version, rtl_fw->version,
                         sizeof(info->fw_version));
 }
 
