--- r8168-8.047.01/src/r8168_n.c~	2019-04-01 17:58:10.000000000 +0200
+++ r8168-8.047.01/src/r8168_n.c	2019-11-30 23:52:41.804500848 +0100
@@ -56,7 +56,7 @@
 #include <linux/rtnetlink.h>
 #include <linux/completion.h>
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)) && (LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0))
 #include <linux/pci-aspm.h>
 #endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,37)
