--- r8168-8.039.00/src/r8168_n.c.orig	2015-04-19 13:32:19.399796241 +0200
+++ r8168-8.039.00/src/r8168_n.c	2015-04-19 13:34:01.866459293 +0200
@@ -3291,9 +3291,12 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3,0,0)
         tag = (tp->vlgrp && vlan_tx_tag_present(skb)) ?
               TxVlanTag | swab16(vlan_tx_tag_get(skb)) : 0x00;
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(4,0,0)
         tag = (vlan_tx_tag_present(skb)) ?
               TxVlanTag | swab16(vlan_tx_tag_get(skb)) : 0x00;
+#else
+        tag = (tp->vlgrp && skb_vlan_tag_present(skb)) ?
+              TxVlanTag | swab16(skb_vlan_tag_get(skb)) : 0x00;
 #endif
 
         return tag;
