--- r8168-8.029.00/src/r8168_n.c~	2012-03-07 09:35:44.000000000 +0100
+++ r8168-8.029.00/src/r8168_n.c	2012-04-06 22:17:27.068049894 +0200
@@ -2229,6 +2229,7 @@
 	.set_settings		= rtl8168_set_settings,
 	.get_msglevel		= rtl8168_get_msglevel,
 	.set_msglevel		= rtl8168_set_msglevel,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,3,0)
 	.get_rx_csum		= rtl8168_get_rx_csum,
 	.set_rx_csum		= rtl8168_set_rx_csum,
 	.get_tx_csum		= rtl8168_get_tx_csum,
@@ -2239,6 +2240,7 @@
 	.get_tso		= ethtool_op_get_tso,
 	.set_tso		= ethtool_op_set_tso,
 #endif
+#endif
 	.get_regs		= rtl8168_get_regs,
 	.get_wol		= rtl8168_get_wol,
 	.set_wol		= rtl8168_set_wol,
